src/theory/strings/core_solver.cpp

```diff
@@ -1137,13 +1137,14 @@ void CoreSolver::processSimpleNEq(NormalForm& nfi,
       // Constants in both normal forms.
       //
       // E.g. "abc" ++ ... = "ab" ++ ...
-      String c1 = x.getConst<String>();
-      String c2 = y.getConst<String>();
+      size_t lenX = Word::getLength(x);
+      size_t lenY = Word::getLength(y);
```

Comment:
Nit: It is a bit confusing to have both `lenX` and `xLen` in this function. Can we maybe rename `xLen` to `xLenN` (length of `x` as a node) and `lenX` to `xLen`? I prefer `xLen` over `lenX` because the different casing of `x` makes things a bit weird.

Reply:
I will rename xLen/yLen to xLenTerm/yLenTerm.